packages/core/useSort/index.ts

```diff
@@ -0,0 +1,63 @@
+import { computed } from '@vue/reactivity'
+import type { MaybeRef } from '@vueuse/shared'
+import { isRef, unref, watchEffect } from 'vue-demi'
+
+export type UseSortCompareFn<T> = (a: T, b: T) => number
+
+export type UseSortFn<T> = (arr: T[], compareFn: UseSortCompareFn<T>) => T[]
+
+export interface UseSortOptions<T> {
+  /**
+   * compare function
+   */
+  compareFn?: UseSortCompareFn<T>
+  /**
+   * change the value of the source array
+   * @default false
+   */
+  dirty?: boolean
+}
+
+const defaultCompare: UseSortCompareFn<number> = (a, b) => a - b
+
+/**
+ * reactive sort array
+ *
+ * @see https://vueuse.org/useSort
+ * @param source source array
+ * @param sortFn sort function
+ * @param options
+ */
+export function useSort<T>(source: MaybeRef<T[]>, sortFn: UseSortFn<T>, options: UseSortOptions<T> = {}) {
```

Comment:
@antfu Perhaps it would be better if I added sortFn to the options, providing a default sort method.

```typescript
export interface UseSortedOptions<T> {
  /**
   * sort algorithm
   */
  sortFn?: UseSortedFn<T>
  /**
   * compare function
   */
  compareFn?: UseSortedCompareFn<T>
  /**
   * change the value of the source array
   * @default false
   */
  dirty?: boolean
}

const defaultSortFn = <T>(source: T[], compareFn: UseSortedCompareFn<T>): T[] => source.sort(compareFn)

export function useSorted<T>(source: MaybeRef<T[]>, options: UseSortedOptions<T> = {}) {
  const { sortFn = defaultSortFn, compareFn = defaultCompare, dirty = false } = options
  // ...
}
```

Reply:
Looks good to me